src/resolve/resolved-varlink.c

```diff
@@ -530,6 +553,516 @@ static int vl_method_resolve_address(Varlink *link, JsonVariant *parameters, Var
         return 1;
 }
 
+static int append_txt(JsonVariant **ret_txt, DnsResourceRecord *rr) {
+        int r;
+
+        assert(rr);
+        assert(rr->key);
+
+        if (rr->key->type != DNS_TYPE_TXT)
+                return 0;
+
+        LIST_FOREACH(items, i, rr->txt.items) {
+                _cleanup_(json_variant_unrefp) JsonVariant *entry = NULL;
+
+                if (i->length <= 0)
+                        continue;
+
+                r = json_variant_new_base64(&entry, i->data, i->length);
+                if (r < 0)
+                        return r;
+
+                r = json_variant_append_array(ret_txt, entry);
```

Comment:
hmm, so now i am confused. our coding style says that you should name return parameters ret_xyz. A return parameter is a purely output param. But here you extend what you get. Hence either the code is wrong or the name (i.e. drop the "ret_" prefix from the parmaete name)

my comment regarding return parameter initialization above is rooted in the misunderstanding that I thought this was a purely output param because of the name, which it appaerntly isn't after all

Reply:
dropped the ret_ prefix.